exonum/src/encoding/serialize/json.rs

```diff
@@ -325,7 +328,7 @@ impl ExonumJson for Vec<RawMessage> {
     ) -> Result<(), Box<dyn Error>> {
         use messages::MessageBuffer;
         let bytes = value.as_array().ok_or("Can't cast json as array")?;
-        let mut vec: Vec<_> = Vec::new();
+        let mut vec: Self = Self::new();
```

Comment:
@SergeiMal Is clippy warning really triggered here? And why are you using `Self` when you set `use_self` to `allow` for this `impl`?

@DarkEld3r If so, seems like here `Self` usage is somewhat confusing. IMHO we should think if we need this lint at all. I like using `Self` in signatures, but this seems too much for me.

Reply:
Also there are some similar cases around the code.

Reply:
This code looks strange for me too.